cxx-checks/src/main/java/org/sonar/cxx/checks/naming/MethodNameCheck.java

```diff
@@ -118,7 +118,7 @@ AstNode getOutsideMemberDeclaration(AstNode declId) {
     if (nestedNameSpecifier != null) {
       AstNode idNode = declId.getLastChild(CxxGrammarImpl.className);
       if (idNode != null) {
-        AstNode className = nestedNameSpecifier.getFirstDescendant(CxxGrammarImpl.className);
+        AstNode className = nestedNameSpecifier.getLastChild(CxxGrammarImpl.className);
```

Comment:
`getLastChild()` addresses the most significant / most outer class name

```
  functionDefinition ==
     [namespaces...]::className0::className1::className2::ID
                                              ^^^^^^^^^^
```

This forces us to express the `nestedNameSpecifier` in terms of one or many repeating `className`s